grpcbp/client_middlewares_test.go

```diff
@@ -0,0 +1,174 @@
+package grpcbp
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"testing"
+	"time"
+
+	pb "github.com/grpc-ecosystem/go-grpc-middleware/testing/testproto"
+	"github.com/opentracing/opentracing-go"
+	"github.com/reddit/baseplate.go/mqsend"
+	"github.com/reddit/baseplate.go/tracing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestMonitorInterceptor(t *testing.T) {
+	// create test server
+	l, _ := initServer(t)
+
+	// create test client with monitoring interceptor
+	conn := initClient(t, l, grpc.WithUnaryInterceptor(
+		MonitorInterceptor(
+			MonitorClientArgs{
+				ServiceSlug: "test",
+			},
+		),
+	))
```

Comment:
creating a test server almost universally implies the need for a test client, and in the few instances where it won't it's nbd to `_` it away.

So have `setupServer` return a client too.

In a past life, we returned a struct that had the client, server, and address in it, so that we could add additional things over time without affecting client code, and that worked really well.